pkg/ddl/partition.go

```diff
@@ -3208,12 +3269,10 @@ func (w *worker) onReorganizePartition(d *ddlCtx, t *meta.Meta, job *model.Job)
 			// Check if they are carried over in ApplyDiff?!?
 			autoIDs, err := t.GetAutoIDAccessors(job.SchemaID, tblInfo.ID).Get()
 			if err != nil {
-				job.State = model.JobStateCancelled
```

Comment:
There is a job error counter that will make it retry (default 512 times I think) and then it will rollback. So I think it is better to keep retrying than fail on the first error.

Reply:
OK, I found the link.

https://docs.pingcap.com/zh/tidb/stable/system-variables#tidb_ddl_error_count_limit